bundles/org.eclipse.cdt.lsp/src/org/eclipse/cdt/lsp/editor/CSpellingReconcileStrategy.java

```diff
@@ -0,0 +1,242 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Bachmann electronic GmbH and others.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ * Gesa Hentschke (Bachmann electronic GmbH) - initial implementation
+ *******************************************************************************/
+
+package org.eclipse.cdt.lsp.editor;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+
+import org.eclipse.cdt.internal.ui.text.spelling.CSpellingService;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.core.runtime.content.IContentType;
+import org.eclipse.core.runtime.content.IContentTypeManager;
+import org.eclipse.jface.text.BadLocationException;
+import org.eclipse.jface.text.IDocument;
+import org.eclipse.jface.text.IRegion;
+import org.eclipse.jface.text.ISynchronizable;
+import org.eclipse.jface.text.Position;
+import org.eclipse.jface.text.Region;
+import org.eclipse.jface.text.reconciler.DirtyRegion;
+import org.eclipse.jface.text.reconciler.IReconcilingStrategy;
+import org.eclipse.jface.text.reconciler.IReconcilingStrategyExtension;
+import org.eclipse.jface.text.source.Annotation;
+import org.eclipse.jface.text.source.IAnnotationModel;
+import org.eclipse.jface.text.source.IAnnotationModelExtension;
+import org.eclipse.jface.text.source.ISourceViewer;
+import org.eclipse.lsp4e.LSPEclipseUtils;
+import org.eclipse.ui.texteditor.spelling.ISpellingProblemCollector;
+import org.eclipse.ui.texteditor.spelling.SpellingAnnotation;
+import org.eclipse.ui.texteditor.spelling.SpellingContext;
+import org.eclipse.ui.texteditor.spelling.SpellingProblem;
+import org.eclipse.ui.texteditor.spelling.SpellingReconcileStrategy;
+import org.eclipse.ui.texteditor.spelling.SpellingService;
+
+/**
+ * Copied from {@link SpellingReconcileStrategy} since we cannot extend it, because have no viewer available at construction time.
```

Comment:
Can we try to use `IExecutableExtensionFactory` to workaround this issue with constructor?

Reply:
I had a deeper look. For this extension point people usually contribute successors of `IReconciler`, not  `IReconcilingStrategy`. However, it may be a lack of API on Platform side.

Also, we already have `CSpellingReconcileStrategy` shouldn't we use it here?

In any case if we don't have appropriate constructor I would rather "decorate" existing spelling strategy, rather than duplicating it.

Reply:
>  I had a deeper look. For this extension point people usually contribute successors of IReconciler

Yes, you are right. I had a failure in my initial  implementation of the CSpellingReconciler. Now it works. Please take a look.